client/src/main/com/sinch/sdk/domains/verification/adapters/WebHooksService.java

```diff
@@ -0,0 +1,89 @@
+package com.sinch.sdk.domains.verification.adapters;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.sinch.sdk.core.exceptions.ApiMappingException;
+import com.sinch.sdk.core.http.AuthManager;
+import com.sinch.sdk.core.utils.Pair;
+import com.sinch.sdk.core.utils.databind.Mapper;
+import com.sinch.sdk.domains.verification.models.webhooks.VerificationEvent;
+import com.sinch.sdk.domains.verification.models.webhooks.VerificationResponse;
+import java.util.Collection;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+
+public class WebHooksService implements com.sinch.sdk.domains.verification.WebHooksService {
+  private static final Logger LOGGER = Logger.getLogger(WebHooksService.class.getName());
+
+  private final Supplier<Map<String, AuthManager>> authManagerSupplier;
+
+  public WebHooksService(Supplier<Map<String, AuthManager>> authManagerSupplier) {
+    this.authManagerSupplier = authManagerSupplier;
+  }
+
+  public boolean checkAuthentication(
+      String method, String path, Map<String, String> headers, String jsonPayload) {
+
+    // convert header keys to use case-insensitive map keys
+    Map<String, String> ciHeaders = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    ciHeaders.putAll(headers);
+
+    String authorizationHeader = ciHeaders.get("Authorization");
+
+    // no authorization required
+    if (null == authorizationHeader) {
+      return true;
+    }
+
+    String[] split = authorizationHeader.split(" ");
+    String authorizationKeyword = split.length > 0 ? split[0] : "";
+    String authorizationHash = split.length > 1 ? split[1] : "";
+
+    // getting content type header
+    String contentTypeHeader = ciHeaders.getOrDefault("content-type", "");
+
+    // getting x-timestamp header
+    String xTimeStampHeader = ciHeaders.get("x-timestamp");
+
+    // getting manager related to Authorization header value
+    AuthManager authManager = authManagerSupplier.get().get(authorizationKeyword);
+
+    // compute locally according to inputs
+    Collection<Pair<String, String>> computedHeaders =
+        authManager.getAuthorizationHeaders(
+            xTimeStampHeader, method, contentTypeHeader, path, jsonPayload);
+
+    // get locally computed hash by auth manager
+    String computedAuthorization =
+        computedHeaders.stream()
+            .filter(f -> f.getLeft().equals("Authorization"))
+            .findFirst()
+            .map(Pair::getRight)
+            .orElse("");
+    String[] newSplit = computedAuthorization.split(" ");
+    String computedHash = newSplit.length > 1 ? newSplit[1] : "";
+
+    return computedHash.equals(authorizationHash);
```

Comment:
This function may be refactored to just be about the comparison between the received against the computed value and put the implementation details about how the extract the data in a private function